client/repo.c

```diff
@@ -545,51 +689,67 @@ TDNFGetRepoMD(
     dwError = TDNFAllocateString(pRepoData->pszId, &pRepoMDRel->pszRepo);
     BAIL_ON_TDNF_ERROR(dwError);
 
+    dwError = TDNFAllocateStringPrintf(&pszBaseUrlFile,
+                                       "%s/%s",
+                                       pszRepoDataDir,
+                                       TDNF_REPO_BASEURL_FILE_NAME);
+    BAIL_ON_TDNF_ERROR(dwError);
     if(metalink)
     {
-        /* if photon.metalink file is not present, set flag to download */
-        if (access(pszMetaLinkFile, F_OK))
+        /* if metalink OR baseurl file is not present, set flag to download */
+        if (access(pszMetaLinkFile, F_OK) || access(pszBaseUrlFile, F_OK))
         {
             if(errno != ENOENT)
             {
                 dwError = errno;
                 BAIL_ON_TDNF_SYSTEM_ERROR(dwError);
             }
+            nStrictlyReplace = 1;
             nNeedDownload = 1;
         }
     }
     else
     {
-        /* if repomd.xml file is not present, set flag to download */
-        if (access(pszRepoMDFile, F_OK))
+        /*If metalink is not set but if metalink and baseurl file is present
+          this applies, metalink was set before but now we are
+          using Base URL instead of Metalink so we need to refresh*/
+
+        if (!access(pszBaseUrlFile, F_OK))
         {
-            if(errno != ENOENT)
+            //we should remove the baseurl file now
+            if(remove(pszBaseUrlFile))
+            {
+                dwError = errno;
+                BAIL_ON_TDNF_SYSTEM_ERROR(dwError);
+            }
+        }
+
+        if (!access(pszMetaLinkFile, F_OK))
+        {
+            //we should remove the metalink file now
+            if(remove(pszMetaLinkFile))
             {
                 dwError = errno;
                 BAIL_ON_TDNF_SYSTEM_ERROR(dwError);
             }
+            /*setting need download true and base url might point to another mirror
+             than the mirror metalink was pointing to */
             nNeedDownload = 1;
+            nStrictlyReplace = 1;
```

Comment:
nStrictlyReplace need not be set here. nDownload handles even if the URL used before from metalink is different from the baseURL which is set now. shasum of repomd files will be different so nDownload will download the repodata again. We can add a separate PR to improvise this logic.